### PR TITLE
Add rate limiter for jira-api calls

### PR DESCRIPTION
**Pull request by guga4ka**
Changes in pr:
1. add `resilience4j-ratelimiter` to implement rate-limiting. It needs to protect Jira from kind of DDoS. I'm planning to add a new pipeline step wich main work will be wrapped in `waitUntil`.

### CODE REVIEW

File: src/main/java/com/atlassian/jira/cloud/jenkins/Config.java [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+import io.github.resilience4j.ratelimiter.RateLimiterRegistry;
+
+import java.time.Duration;
+
 public interface Config {
 
     String ATLASSIAN_API_URL = "https://api.atlassian.com";
+    String ATLASSIAN_RATE_LIMITER_CONFIG = "atl";
+
+    RateLimiterConfig DEFAULT_RATE_LIMITER_CONFIG =
```
Comment (guga4ka) [inline src/main/java/com/atlassian/jira/cloud/jenkins/Config.java:14]: This isn't final config. I've used [these](https://developer.atlassian.com/cloud/jira/platform/user-impersonation-for-connect-apps/#rate-limiting) limits as an example

File: pom.xml [hunk 2]
```diff
@@ -25,6 +25,8 @@
         <assertj.version>3.11.1</assertj.version>
         <maven-checkstyle-plugin.version>2.17</maven-checkstyle-plugin.version>
         <archunit.version>0.9.2</archunit.version>
+        <resilience4j.version>1.4.0</resilience4j.version>
+        <slf4j.version>1.7.26</slf4j.version>
```
Comment (guga4ka) [inline pom.xml:29]: Explicitly define `slf4j` version needed for `resilience4j`

**Review by aolrich323 — APPROVED** [review]
These changes look good to me from the perspective of suitability with the JSD change management feature requirements.

File: src/main/java/com/atlassian/jira/cloud/jenkins/common/client/JiraApi.java [hunk 3]
```diff
@@ -72,6 +73,8 @@ public JiraApi(
                             "Server exception when submitting update to Jira: %s", e.getMessage()));
         } catch (ApiUpdateFailedException e) {
             return handleError(e.getMessage());
+        } catch (RequestNotPermitted e) {
+            return handleError("Your OAuth client riches Jira's limits " + e.getMessage());
```
Comment (lenin-atlassian) [inline src/main/java/com/atlassian/jira/cloud/jenkins/common/client/JiraApi.java:77]: `riches` - typo?

**Review by lenin-atlassian — APPROVED** [review]

**Pull request merged**